Tribler/Test/Core/Modules/RestApi/test_my_channel_endpoints.py

```diff
@@ -220,6 +232,182 @@ def test_rss_feeds_endpoint_no_my_channel(self):
+
+        torrent_file = open(torrent_path, mode='rb')
+        torrent_64 = base64.b64encode(torrent_file.read())
+        post_data = {
+            "torrent": torrent_64
+        }
+        self.should_check_equality = False
+        return self.do_request('mychannel/torrents', expected_code=500, expected_json=None, request_type='PUT',
+                               post_data=post_data).addCallback(verify_error_message)
+
+
+class TestMyChannelModifyTorrentsEndpoint(AbstractTestMyChannelEndpoints):
+
+    def setUpPreSession(self):
+        super(TestMyChannelModifyTorrentsEndpoint, self).setUpPreSession()
+        self.config.set_libtorrent(True)
+
+    @deferred(timeout=10)
+    def test_add_torrent_from_url_to_my_channel_with_description(self):
+        my_channel_id = self.create_fake_channel("channel", "")
+        my_torrent_def = TorrentDef.load(join(TESTS_DATA_DIR, 'archlinux-2016.05.01-dual.iso.torrent'))
+
+        @staticmethod
+        def fake_load_from_url(url):
+            return my_torrent_def
+
+        TorrentDef.load_from_url = fake_load_from_url
+
+        def verify_method_invocation(channel_id, torrent_def, extra_info={}, forward=True):
+            self.assertEqual(my_channel_id, channel_id)
+            self.assertEqual(my_torrent_def, torrent_def)
+            self.assertEqual({"description": "archlinux 2016.05.01"}, extra_info)
+            self.assertEqual(True, forward)
+
+        self.session.add_torrent_def_to_channel = verify_method_invocation
+
+        torrent_url = 'https://www.archlinux.org/releng/releases/2016.05.01/torrent/'
+        url = 'mychannel/torrents/' + urllib.quote_plus(torrent_url)
+        return self.do_request(url, expected_code=200, expected_json={"added": torrent_url}, request_type='PUT',
+                               post_data={"description": "archlinux 2016.05.01"})
+
+    @deferred(timeout=10)
+    def test_add_torrent_from_magnet_to_my_channel_without_description(self):
+        my_channel_id = self.create_fake_channel("channel", "")
+        my_torrent_def = TorrentDef.load(join(TESTS_DATA_DIR, 'archlinux-2016.05.01-dual.iso.torrent'))
+
+        def fake_load_from_dht(infohash_or_magnet, callback):
+            meta_info = my_torrent_def.get_metainfo()
+            callback(meta_info)
+
+        self.session.lm.ltmgr.get_metainfo = fake_load_from_dht
+
+        def verify_method_invocation(channel_id, torrent_def, extra_info={}, forward=True):
+            self.assertEqual(my_channel_id, channel_id)
+            self.assertEqual(my_torrent_def, torrent_def)
+            self.assertEqual({}, extra_info)
+            self.assertEqual(True, forward)
+
+        self.session.add_torrent_def_to_channel = verify_method_invocation
+
+        magnet_url = 'magnet:?xt=urn:btih:33130de5c14a8bb5410746ee5a9604cdfb9538ef&dn=archlinux-2016.05.01-dual.iso&tr=udp://tracker.archlinux.org:6969&tr=http://tracker.archlinux.org:6969/announce'
```

Comment:
Is there a reason we aren't using the torrent constants we have already in place and used in the other tests?
(TORRENT_FILE & co.)